plugins/sqs/sqs.cpp

```diff
@@ -0,0 +1,539 @@
+
+  Aws::SQS::Model::ReceiveMessageOutcome receiveMessageOutcome = this->sqsClient->ReceiveMessage(receiveMessageRequest);
+ if(!receiveMessageOutcome.IsSuccess() || receiveMessageOutcome.GetResult().GetMessages().size() == 0) {
+       std::cout << "Error on receive: " << receiveMessageOutcome.GetError().GetMessage() << std::endl;
+       ref.code=2;
+       return ref;
+ }
+ 
+ Aws::SQS::Model::Message msg = receiveMessageOutcome.GetResult().GetMessages()[0];
+ cout << msg.GetBody() << endl;
+} catch(const char* message) {
+
+  cout << "Error occurs " << message << endl; 
+}
+
+ return ref;
+}
+
+
+/**
+ *
+ *
+ *
+ *
+ **/
+void SQSHPCCPlugin::SQSHPCC::setQueueUrlFromQueueName() 
+{
+
+   Aws::SQS::Model::GetQueueUrlRequest* gqu_req = new Aws::SQS::Model::GetQueueUrlRequest();
+   gqu_req->SetQueueName(this->queueName.c_str());
+
+try {
+   Aws::SQS::Model::GetQueueUrlOutcome gqu_out = this->sqsClient->GetQueueUrl(*gqu_req);
+//const char* bl = gqu_out.IsSuccess() ? "true":"false";
+//Aws::String strBl(bl);
+//this->queueUrl=strBl;
+  if(gqu_out.IsSuccess()) {
+        std::cout << "Queue " << this->queueName.c_str() << " has url " << std::endl;
+    this->queueUrl=gqu_out.GetResult().GetQueueUrl();
+   } else {
+        std::cout << "Error getting url for queue " << this->queueName.c_str() << ": " << std::endl;
+      throw runtime_error(gqu_out.GetError().GetMessage().c_str()) ;
+    // this->queueUrl = gqu_out.GetError().GetMessage();
+   }
+
+ } catch(const char* message) {
+     cout << "Error occurs " << message << endl;
+}
+
+ delete(gqu_req);
+}
+
+
+/**
+
+
+ **/
+void SQSHPCCPlugin::SQSHPCC::setSQSConfiguration(const string& protocol, const string& region)
+ {
+
+  Aws::InitAPI(this->options);
+  Aws::Client::ClientConfiguration* config= new Aws::Client::ClientConfiguration();
+ if(!protocol.empty())
+  {
+    config->scheme = Aws::Http::Scheme::HTTPS;
+  }
+
+  if(region.empty()) throw string("Region mustn't be empty");
+  if(isRegionExist(region))
+    {
+      config->region = getRegion(region);
+    }
+
+  this->sqsClient= new Aws::SQS::SQSClient(*config);
+
+ //this->queueUrl=getRegion(region);
+   cout << "Queue URL is "  << this->queueUrl << endl;
+// this->setQueueUrlFromQueueName();
+ 
+ delete(config);
+}
+
+
+
+/**
+
+
+ **/
+bool SQSHPCCPlugin::SQSHPCC::isRegionExist(const string& region)
+{
+  
+  if(region.empty())
+    {
+      return false;
+    }
+
+  char *reg = new char[region.length() +1];
+  strcpy(reg,region.c_str());
+  upstr(reg);
+ 
+  if((strncmp(reg,"US_EAST_1",9)==0)|| (strncmp(reg,"US_WEST_1",9)==0) || (strncmp(reg,"EU_WEST_1",9)==0)
+      || (strncmp(reg,"EU_CENTRAL_1",12)==0) || (strncmp(reg,"AP_SOUTHEAST_1",14)==0) || (strncmp(reg,"AP_SOUTHEAST_2",14)==0)
+     ) return true;
+  
+  return false;
+  
+}
+
+
+/**
+*
+*
+*/
+const char *const SQSHPCCPlugin::SQSHPCC::getRegion(const string& region) 
+{
+
+  char *reg = new char[region.length() +1];
```

Comment:
Similar comments to above (and not sure why the code above doesn't just call this method and return whether the value was NOTFOUND or not...)